kitsune/dashboards/views.py

```diff
@@ -252,10 +252,10 @@ def aggregated_metrics(request):
 
 def _get_product(request):
     product_slug = request.GET.get("product")
-    if product_slug:
```

Comment:
Hi @gauravghati and thanks for this PR. May I ask why are proposing this change?

Reply:
Yes, the `request.GET.get('product')` is returning `[]` as a string, when the "all product" option is selected from the dropdown menu.

So, that's why, funtion `get_object_or_404` is returing page 404 when `product_slug="[]"`

So, putting it in try-except will return `product=None` if `product_slug="[]"`, that will help us to remove the filter condition from [line#222](https://github.com/mozilla/kitsune/blob/eb32f145c5b9095fcea75fd7f1341604bca35b69/kitsune/dashboards/readouts.py#L222) in readouts.py. Removing the filter condition will return all products from the database.